src/client/array/dc_array.c

```diff
@@ -2356,32 +2355,50 @@ add_record(daos_handle_t oh, daos_handle_t th, struct set_size_props *props)
 	if (rc)
 		D_GOTO(err, rc);
 
-	rc = tse_task_schedule(io_task, false);
-	if (rc)
-		D_GOTO(err, rc);
-
+	/* decref in adjust_array_size_task_process() */
+	tse_task_addref(io_task);
+	tse_task_list_add(io_task, task_list);
 	return rc;
+
 err:
 	D_FREE(params);
 	if (io_task)
 		tse_task_complete(io_task, rc);
 	return rc;
 }
 
+static int
+adjust_array_size_task_process(tse_task_t *task, void *arg)
```

Comment:
this seems should be in tse.c?

Reply:
it is tse user's task list's task, seems fine here.